main.go

```diff
@@ -119,18 +142,31 @@ func markPDF(inputPath string, outputPath string, watermark string) error {
 			drawImage(watermarkImg, c)
 
 		} else {
-			if pageNum == 1 {
-				para = creator.NewParagraph(watermark)
-				adjustTextPosition(para, c)
-			}
+
+			// Execute the template for each page.
+			buf.Reset()
+			err := t.Execute(buf, rec)
+			fatalIfError(err, fmt.Sprintf("Failed to execute watermark template: [%s]", err))
+
+			para = creator.NewParagraph(buf.String())
+			adjustTextPosition(para, c)
```

Comment:
I could not benchmark yet, but my feelings is - preparing the static content (only once) would be faster than executing template `t.Execute(buf, rec)` for every page. Specially, when working with PDFs with 300+ pages, it will make a difference.

So, can we do this?

1. check the `watermark` text to identify if placeholder variables are being used
2. Prepare `Recipient` and parse the template only if it's required
3. At this point, let's keep older static text based one time `para = creator.NewParagraph(watermark)` unless placeholders are used.
4. Similar stretegy for writing output file.